worker/processors/playerTracking/index.js

```diff
@@ -0,0 +1,118 @@
+    server: server.id,
+    steamId: onlinePlayers.map(playerInfo => playerInfo.steamid)
+  });
+
+  for (const playerRecord of playerRecords) {
+
+    initialValues.push({
+      server: server.id,
+      player: playerRecord.id
+    });
+  }
+
+  try {
+    await trackingFunctions.basic(server, onlinePlayers, playerRecords);
+  } catch (error) {
+    sails.log.error(error);
+  }
+
+  if (server.config[0].locationTracking || server.config[0].inventoryTracking) {
+    // If inventory OR location tracking is enabled, we prepare the tracking info beforehand to improve performance
+
+    if (server.config[0].locationTracking) {
+
+      try {
+        initialValues = await trackingFunctions.location(server, onlinePlayers, initialValues, playerRecords);
+      } catch (error) {
+        sails.log.error(error);
+      }
+    }
+
+    if (server.config[0].inventoryTracking) {
+
+      try {
+        initialValues = await trackingFunctions.inventory(server, onlinePlayers, initialValues, playerRecords);
+      } catch (error) {
+        sails.log.error(error);
+      }
+      await sails.helpers.getQueueObject('bannedItems').add({ server, trackingInfo: initialValues });
+    }
+    await TrackingInfo.createEach(initialValues);
+
+  }
+
+  let currentCycles = await sails.helpers.redis.get(`server:${serverId}:trackingCyclesCompleted`);
+  currentCycles = parseInt(currentCycles);
+
+  if (!currentCycles) {
+    currentCycles = 1;
+  }
+
+  if (currentCycles >= sails.config.custom.trackingCyclesBeforeDelete) {
+    await deleteLocationData(server);
+    await sails.helpers.redis.set(`server:${serverId}:trackingCyclesCompleted`, 0);
+  } else {
+    await sails.helpers.redis.incr(`server:${serverId}:trackingCyclesCompleted`);
+  }
+
+  let dateEnded = new Date();
+  sails.log.debug(`Player tracking - Performed tracking for server ${server.name} - ${playerRecords.length} players online - ${currentCycles}/${sails.config.custom.trackingCyclesBeforeDelete} tracking cycles - took ${dateEnded.valueOf() - dateStarted.valueOf()} ms`);
+
+}
+
+
+async function deleteLocationData(server) {
+  let dateNow = Date.now();
+  let deleteResult;
+  try {
+    let donatorRole = await sails.helpers.meta.checkDonatorStatus.with({
+      serverId: server.id
+    });
+
+    let hoursToKeepData = sails.config.custom.donorConfig[donatorRole].playerTrackerKeepLocationHours;
+    let milisecondsToKeepData = hoursToKeepData * 3600000;
+    let borderDate = new Date(dateNow.valueOf() - milisecondsToKeepData);
+
+    const locationDeleteSQL = `DELETE FROM trackinginfo WHERE server = ${server.id} AND createdAt < ${borderDate.valueOf()};`;
+
+    deleteResult = await sails.sendNativeQuery(locationDeleteSQL);
+
+  } catch (error) {
+    sails.log.error(error);
+  }
+
+  let dateEnded = new Date();
+  sails.log.verbose(`Deleted location data for server ${server.name} - deleted ${deleteResult.affectedRows} rows - took ${dateEnded.valueOf() - dateNow.valueOf()} ms`);
```

Comment:
if there's an error, deletedResult will be null (since its the last statement) which crashes

I personally would remove the try/catch entirely and let upstream things handle it, but all the lets should be consts, and everything should be inside the try.